apimodels/conversions.go

```diff
@@ -68,9 +80,49 @@ func ConvertBlockModelToBlockResponse(block *dbmodels.Block) *BlockResponse {
 	}
 	if block.AcceptingBlock != nil {
 		blockRes.AcceptingBlockHash = &block.AcceptingBlock.BlockHash
+		blockRes.AcceptingBlockBlueScore = &block.AcceptingBlock.BlueScore
 	}
+	blockRes.Confirmations = pointers.Uint64(confirmations(blockRes.AcceptingBlockBlueScore, selectedTipBlueScore))
 	for i, parent := range block.ParentBlocks {
 		blockRes.ParentBlockHashes[i] = parent.BlockHash
 	}
 	return blockRes
 }
+
+// ConvertTransactionOutputModelToTransactionOutputResponse converts a transaction output
+// database object into a TransactionOutputResponse
+func ConvertTransactionOutputModelToTransactionOutputResponse(transactionOutput *dbmodels.TransactionOutput,
+	selectedTipBlueScore uint64, activeNetParams *dagconfig.Params, isSpent bool) (*TransactionOutputResponse, error) {
+	subnetworkID := &subnetworkid.SubnetworkID{}
+	err := subnetworkid.Decode(subnetworkID, transactionOutput.Transaction.Subnetwork.SubnetworkID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "couldn't decode subnetwork id %s", transactionOutput.Transaction.Subnetwork.SubnetworkID)
+	}
+	var acceptingBlockHash *string
+	var acceptingBlockBlueScore *uint64
+	if transactionOutput.Transaction.AcceptingBlock != nil {
+		acceptingBlockHash = &transactionOutput.Transaction.AcceptingBlock.BlockHash
+		acceptingBlockBlueScore = &transactionOutput.Transaction.AcceptingBlock.BlueScore
+	}
+	isCoinbase := subnetworkID.IsEqual(subnetworkid.SubnetworkIDCoinbase)
+	utxoConfirmations := confirmations(acceptingBlockBlueScore, selectedTipBlueScore)
+
+	txOutRes := &TransactionOutputResponse{
+		TransactionID:           transactionOutput.Transaction.TransactionID,
+		Value:                   transactionOutput.Value,
+		ScriptPubKey:            hex.EncodeToString(transactionOutput.ScriptPubKey),
+		AcceptingBlockHash:      acceptingBlockHash,
+		AcceptingBlockBlueScore: acceptingBlockBlueScore,
+		Index:                   transactionOutput.Index,
+		IsCoinbase:              &isCoinbase,
+		Confirmations:           &utxoConfirmations,
+	}
+
+	isSpendable := false
```

Comment:
Better to first calculate `isSpendable`, and only than instantiate `txOutRes`, with isSpendable already known, just as we do with `isCoinbase` and `utxoConfirmation`